### ISSUE TITLE
Support Apache 2.4

### ISSUE THREAD

**Issue by esabol** [matisse/Apache-AuthCookieDBI#2]
It would be really nice if AuthCookieDBI supported Apache 2.4.

See http://search.cpan.org/~mschout/Apache-AuthCookie/README.apache-2.4.pod for details.

Reference: https://rt.cpan.org/Public/Bug/Display.html?id=106663

**Comment by esabol** [comment]
Hello? Is anyone there? :(

**Comment by matisse** [comment]
Yes

**Comment by esabol** [comment]
That’s good to know. 👍

Any thoughts on support for Apache 2.4 in AuthCookieDBI?

**Comment by matisse** [comment]
As you can probably tell I haven’t done active development on the module in a long time can you say something about what the issues with Apache 2.4 are?

**Comment by esabol** [comment]
Group-based authorization doesn't work. Under Apache 2.4, one needs to specify an AuthzProvider for group that handles checking for group membership., like this:
```
PerlAddAuthzProvider group Apache2::AuthCookieDBI->authz_group
```
This hypothetical authz_group() method would be similar to AuthCookieDBI's group() method, but it needs to return specific Apache 2.4+ constants and handle things somewhat differently. Again, this is largely documented at

http://search.cpan.org/~mschout/Apache-AuthCookie/README.apache-2.4.pod

There's also a thread about this here:

http://mail-archives.apache.org/mod_mbox/perl-modperl/201902.mbox/browser

**Comment by matisse** [comment]
Would you be interested in creating a pull request?

**Comment by esabol** [comment]
> Would you be interested in creating a pull request?

If I can get access to a system running Apache 2.4 + mod_perl that I could develop on and can get AuthCookieDBI working, then I surely will.

**Comment by matisse** [comment]
Released version 2.19 today https://metacpan.org/release/Apache2-AuthCookieDBI

**Comment by esabol** [comment]
Wonderful! Thank you very much! Cheers!
